python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,266 @@
+import unittest
+import warnings
+import pandas as pd
+import numpy as np
+from sklearn import datasets
+from sklearn.linear_model import Perceptron, LogisticRegression
+from ml4ir.applications.ranking.pipeline import RankingPipeline
+from ml4ir.applications.ranking.config.parse_args import get_args
+import shutil
+import pathlib
+import random
+from argparse import Namespace
+warnings.filterwarnings("ignore")
+
+
+def create_ml4ir_dataframe(X, Y):
+    """
+    Takes the generated data (X: data features, Y: clicked) and formulate a dataframe consumable by ml4ir
+    """
+    def add_features_to_record(X, xid, r, label):
+        for i in range(len(X[xid])):
+            r['f' + str(i)] = X[xid][i]
+        r['clicked'] = label
+
+    clicked_docs = X[Y == 1]
+    nonclicked_docs = X[Y == 0]
+    consumed_clicked_docs = 0
+    consumed_nonclicked_docs = 0
+    rows = []
+    query_count = 0
+    document_count = 0
+    while consumed_clicked_docs < len(clicked_docs) and consumed_nonclicked_docs < len(nonclicked_docs):
+        qid = 'query'+str(query_count)
+        query_count += 1
+        r = {'query_id':qid}
+        docs_count = random.randint(2, 25)
+        r['record_id'] = 'record'+str(document_count)
+        r['fr'] =docs_count
+        add_features_to_record(clicked_docs, consumed_clicked_docs, r, 1)
+        consumed_clicked_docs += 1
+        document_count += 1
+        rows.append(r)
+        for i in range(docs_count-1):
+            if consumed_nonclicked_docs < len(nonclicked_docs):
+                r = {'query_id': qid}
+                r['fr'] = i+1
+                r['record_id'] = 'record' + str(document_count)
+                add_features_to_record(nonclicked_docs, consumed_nonclicked_docs, r, 0)
+                consumed_nonclicked_docs += 1
+                document_count += 1
+                rows.append(r)
+            else:
+                break
+    return pd.DataFrame(rows)
+
+
+def generate_data(args):
```

Comment:
I get the temptation to do this and I've done it in the past as well, but I think listing all arguments individually here is a more robust coding patten. @mohazahran

Reply:
You should also definitely add default values for all these arguments, maybe the same as the ones in sklearn if that's the best for the test.